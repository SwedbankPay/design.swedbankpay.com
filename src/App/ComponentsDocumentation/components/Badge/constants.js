import React from "react";

const BadgeOptions = ({ number }) => (
	<>
		{number ? (
			<div className="badge-showcase">
				<p>
					{" "}
					<span className="badge badge-default badge-number">1</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-inactive badge-number">2</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-warning badge-number">3</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-yellow badge-number">4</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-gray badge-number">5</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-red badge-number">6</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-blue badge-number">7</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-pink badge-number">8</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-green badge-number">9</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-orange badge-number">10</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-turqouise badge-number">11</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-brown badge-number">12</span>
				</p>
			</div>
		) : (
			<div className="badge-showcase">
				<p>
					{" "}
					<span className="badge badge-default">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-inactive">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-warning">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-yellow-light">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-yellow-lighter">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-gray">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-gray-light">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-brown">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-brown-soft">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-pink">Document</span>
				</p>
				<p>
					{" "}
					<span className="badge badge-red-light">Document</span>
				</p>
			</div>
		)}
	</>
);

export const badgeShowcase = {
	id: "no-tabs",
	elements: [
		{
			component: <BadgeOptions id="badge-showcase" />,
			options: {
				radio: [
					{
						id: "badge_type_radio",
						title: "Badge Type",
						values: [
							{
								id: "radio-text",
								name: "Text",
								value: {
									number: false,
								},
							},
							{
								id: "radio-numbers",
								name: "Number",
								value: {
									number: true,
								},
							},
						],
					},
				],
			},
			title: "Badge",
			description:
				"Use the Badge Component Symbol to draw attention to another interface element or to display a notification. It enhances the component to which it is attached with additional information, disclosed to the user upon interacting with it.",
		},
	],
};
