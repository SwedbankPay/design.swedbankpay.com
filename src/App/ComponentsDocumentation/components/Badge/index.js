import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import CodeTags from "@components/CodeTags";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Use class <CodeTags type="secondary" code=".badge" /> along with <CodeTags type="secondary" code=".badge-{color}" /> to display the badge of your choosing.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p>Badge <span className="badge badge-default">Default</span></p>
            <p>Badge <span className="badge badge-inactive">Inactive</span></p>
            <p>Badge <span className="badge badge-warning">Warning</span></p>
        </ComponentPreview>
    </>
);

const BadgeNumber = () => (
    <>
        <h2 id="badgenumber">Badges with numbers</h2>
        <p>Badges are often used with numbers. Use <CodeTags type="secondary" code=".badge-number" /> when a badge suited for use with numbers is desired.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p>Badge <span className="badge badge-default badge-number">1</span></p>
            <p>Badge <span className="badge badge-inactive badge-number">8</span></p>
            <p>Badge <span className="badge badge-warning badge-number">100</span></p>
        </ComponentPreview>
    </>
);

const Badge = () => (
    <DocContainer docToc>
        <p className="lead">
            Badges are small rectangular boxes which can convey a message and or state with their text and color.
        </p>
        <Overview />
        <BadgeNumber />
    </DocContainer>
);

export default Badge;

/* For testing */
export { Overview, BadgeNumber };
