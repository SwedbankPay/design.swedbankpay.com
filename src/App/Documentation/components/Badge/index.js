import React from "react";

import { ComponentPreview, Property, DocContainer } from "@docutils";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Use class <Property value=".badge" /> along with <Property value=".badge-{color}" /> to display the badge of your choosing.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p>Badge <span className="badge badge-default">default</span></p>
            <p>Badge <span className="badge badge-yellow">yellow</span></p>
            <p>Badge <span className="badge badge-blue">blue</span></p>
            <p>Badge <span className="badge badge-red">red</span></p>
        </ComponentPreview>
    </>
);

const BadgeNumber = () => (
    <>
        <h2 id="badgenumber">Badges with numbers</h2>
        <p>Badges are often used with numbers. Use <Property value=".badge-number" /> when a badge suited for usage with numbers is desired.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p>Badge <span className="badge badge-default badge-number">1</span></p>
            <p>Badge <span className="badge badge-yellow badge-number">8</span></p>
            <p>Badge <span className="badge badge-yellow badge-number">42</span></p>
            <p>Badge <span className="badge badge-blue badge-number">100</span></p>
        </ComponentPreview>
    </>
);

const Badge = () => (
    <DocContainer docToc>
        <p className="lead">
            Badges are small rectangular boxes which can convey a message and or state with their text and color.
        </p>
        <Overview />
        <BadgeNumber />
    </DocContainer>
);

export default Badge;

/* For testing */
export { Overview, BadgeNumber };
