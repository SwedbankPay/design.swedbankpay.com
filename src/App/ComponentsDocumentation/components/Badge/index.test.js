import React from "react";
import { shallow } from "enzyme";

import Badge, { Overview } from "./index";

describe("Components: Badge", () => {
    it("is defined", () => {
        expect(Badge).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Badge />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
