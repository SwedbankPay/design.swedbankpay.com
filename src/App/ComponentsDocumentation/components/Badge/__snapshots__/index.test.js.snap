// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Badge Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Use class 
    <CodeTags
      code=".badge"
      type="secondary"
    />
     along with 
    <CodeTags
      code=".badge-{color}"
      type="secondary"
    />
     to display the badge of your choosing. Badges are often used with numbers. Use 
    <CodeTags
      code=".badge-number"
      type="secondary"
    />
     when a badge suited for use with numbers is desired.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
    showCasePanelAdvanced={
      Object {
        "elements": Array [
          Object {
            "component": <BadgeOptions
              id="badge-showcase"
            />,
            "description": "Use the Badge Component Symbol to draw attention to another interface element or to display a notification. It enhances the component to which it is attached with additional information, disclosed to the user upon interacting with it. ",
            "options": Object {
              "radio": Array [
                Object {
                  "id": "badge_type_radio",
                  "title": "Badge Type",
                  "values": Array [
                    Object {
                      "id": "radio-text",
                      "name": "Text",
                      "value": Object {
                        "number": false,
                      },
                    },
                    Object {
                      "id": "radio-numbers",
                      "name": "Number",
                      "value": Object {
                        "number": true,
                      },
                    },
                  ],
                },
              ],
            },
            "title": "Badge",
          },
        ],
        "id": "no-tabs",
      }
    }
  />
  <h3>
    Things to consider
  </h3>
  <ul
    className="list list-bullet"
  >
    <li>
      A normal badge does not feature interaction, use the button component instead.
    </li>
    <li>
      If you want removable badges considering using the tag component instead.
    </li>
    <li>
      Do not use too long labels or whole sentences.
    </li>
    <li>
      If possible, try not to use more than three words.
    </li>
  </ul>
  <h2>
    Do's and Dont's
  </h2>
  <div
    className="row"
  >
    <div
      className="col-lg-6"
    >
      <div
        className="slab slab-plain slab-border-success"
      >
        <h3>
          Do
        </h3>
        <div
          className="d-flex justify-content-center mt-4 mb-4"
        >
          <span
            className="badge badge-yellow-light"
          >
            Transactions
          </span>
        </div>
      </div>
    </div>
    <div
      className="col-lg-6"
    >
      <div
        className="slab slab-plain slab-border-error"
      >
        <h3>
          Don't
        </h3>
        <div
          className="d-flex justify-content-center mt-4 mb-4"
        >
          <span
            className="badge badge-yellow-light"
          >
            Transactions from 2022-2023
          </span>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Components: Badge renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Badges are small rectangular boxes which can convey a message and or state with their text and color.
  </p>
  <Overview />
</DocContainer>
`;
